### PR TITLE
Remove arcjs dep

### PR DESCRIPTION
**Pull request by shivgupt**
Update documentation and Replace Arc.js with Client and Subgraph

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@
 - [The Stack](#the-stack)
   - [Infra](#infra)
   - [Arc](#arc)
-  - [Arc.js](#arcjs)
+  - [Client](#client)
```
Comment (orenyodfat) [inline README.md]: where does subgraph is getting in ?
the client is a layer on top the subgraph

Comment (shivgupt) [reply README.md]: So I briefly talked about subgraph in the Client section. Should I remove it or are you suggesting to mention that as part of Stack?

Comment (ben-kaufman) [reply README.md]: I think this should be mentioned as a part of the stack, though you're right it's strongly connected to the client.

File: starter-template/README.md [hunk 2]
```diff
@@ -10,97 +10,60 @@ You can then go ahead and edit the template to fit your needs.
...
-After downloading the project:
+- copy the `example.env` from `data` directory as `.env` in the project directory
+- Update the `.env` file with your `SEED_PHRASE` and <TESTNET-NAME>`_INFURA_URL`
+- Enter the project folder from the terminal (starter-template) and type the following:
 
 ```
-npm install
-npm install -g nps
-npm install -g npx
+npm run launch:docker
```
Comment (ben-kaufman) [inline starter-template/README.md]: Maybe it's too obvious but we can just remind here to run `npm install`.

File: starter-template/README.md [hunk 3]
```diff
@@ -10,97 +10,60 @@ You can then go ahead and edit the template to fit your needs.
...
-```
-rm -rf build
-truffle migrate --network kovan-infura
-```
+### Deploy and use on test network:
 
-5. Copy the Avatar and Voting Machine addresses from the end of the `truffle migrate` logs.
-6. Open `src/index.js` and pasted the copied addresses in the `avatarAddress` and `votingMachineAddress`.
-7. On the terminal window, run the following command:
+- Update the `.env` file with your `SEED_PHRASE` and <TESTNET-NAME>`_INFURA_URL`
```
Comment (ben-kaufman) [inline starter-template/README.md]: <TESTNET-NAME>`_INFURA_URL` -> `<TESTNET-NAME>_INFURA_URL`
(just to make it look nicer in markdown)

File: starter-template/README.md [hunk 4]
```diff
@@ -10,97 +10,60 @@ You can then go ahead and edit the template to fit your needs.
...
-```
-
-### Deploy and use on Kovan network:
+In another browser window/tab type *http://localhost:8000/subgraphs/name/daostack* This is an interface to the subgraph and you can type graphQL queries here to fetch data from the `postgres` database
 
-1. Enter the `truffle.js` file and make the changes as instructed there.
-2. Open `src/index.js` and uncomment the lines to configure Kovan (as instructed there).
-3. Open terminal at the project folder
-4. Run the following commands:
+To learn more about subgraphs check out: [DAOstack Subgraph](https://github.com/daostack/subgraph), [TheGraph Project](https://thegraph.com/docs/quick-start)
```
Comment (ben-kaufman) [inline starter-template/README.md]: Please just add somewhere that to stop the project you should run `npm run stop`.

File: README.md (hunk 1 above)
Comment (orenyodfat) [reply README.md]: subgraph (our indexing caching layer) is part of our stack.
one can build its own client on top of the graph service .

File: README.md [hunk 5]
```diff
@@ -61,7 +61,7 @@ DAOs built with Arc have a few basic contract components:
...
 
-**Arc utilizes the concept of "Universal" contracts**: contracts which are deployed once, and then can be used by any number of DAOs simultaneously, saving gas and deployment costs. Schemes and constraints can both be used in this way. To use the already deployed contracts, you can either use Arc.js, which maintains easy access to all universal Arc contracts, or you can use [this table](https://docs.google.com/spreadsheets/d/1hfR-fnnqXEn3Go3x3qoiXJcKSIAYQalSOi52vV2PCTA/edit?usp=sharing) to view the addresses of the universal contracts on the mainnet, Kovan, and Ganache\*
+**Arc utilizes the concept of "Universal" contracts**: contracts which are deployed once, and then can be used by any number of DAOs simultaneously, saving gas and deployment costs. Schemes and constraints can both be used in this way. To use the already deployed contracts, you can either use Client, which maintains easy access to all universal Arc contracts, or you can use [this table](https://docs.google.com/spreadsheets/d/1hfR-fnnqXEn3Go3x3qoiXJcKSIAYQalSOi52vV2PCTA/edit?usp=sharing) to view the addresses of the universal contracts on the mainnet, Kovan, and Ganache\*
```
Comment (orenyodfat) [inline README.md]: I guess that the google doc should be replaced with https://github.com/daostack/migration/blob/master/migration.json

File: README.md [hunk 6]
```diff
@@ -72,31 +72,30 @@ Using Arc is not necessary to deploy a DAO (you can do this with Arc.js currentl
...
-
-You can find NodeJs scripts that use Arc.js to perform a variety of functions in the [Arc.js-scripts Github repository](https://github.com/daostack/arc.js-scripts).
+## [Subgraph](https://github.com/daostack/subgraph)
+**Subgraph indexes the blockchain data and stores it in postgres database for easy and quich access.** The data store can be queried by GraphQL endpoints. DAOstack subgraph is based on the graphprotocol checkout [TheGraph](https://thegraph.com/docs/quick-start) for more details.
 
 ### Should I work at this level?
 
-You should use Arc.js whenever you want to use JavaScript or TypeScript to interact with Arc contracts. This is particularly helpful for developers who want to get the advantages of decentralized governance on the blockchain without dealing directly with a smart contract language.
+You should use Client whenever you want to use JavaScript or TypeScript to interact with Arc contracts. This is particularly helpful for developers who want to get the advantages of decentralized governance on the blockchain without dealing directly with a smart contract language.
```
Comment (orenyodfat) [inline README.md]: this part is related to subgraph . no ?
so : 
"You should use Subgraph service whenever you want to query our platform re  Dao states/proposals... and so forth.."

**Comment by orenyodfat** [comment]
- do we need the whole subgraph code in the starter example ?
- where does the migration repo is mention ? how to deploy a single Dao ?

**Comment by ben-kaufman** [comment]
Another thing to add is that because the subgraph repository is regularly updated the folder should be maintained with the changs.

File: README.md (hunk 1 above)
Comment (shivgupt) [reply README.md]: Gotcha 👍🏻

File: starter-template/README.md (hunk 2 above)
Comment (shivgupt) [reply starter-template/README.md]: Oh yeah right now I have only added Instructions for running inside docker, In which case `npm install` is run while building docker image and not needed separately.
But I can add another section with instructions to run without docker.

File: README.md [hunk 7]
```diff
@@ -72,31 +72,30 @@ Using Arc is not necessary to deploy a DAO (you can do this with Arc.js currentl
...
 
-Should you chose to leverage Arc.js in your application, you can find helpful documentation in the [complete Arc.js documentation](https://daostack.github.io/arc.js/).
+Using Client, JavaScript/TypeScript developers can easily write scripts or applications which can interact with existing DAOs, submit proposals to DAOs, vote and stake on proposals, execute the resulting decisions, manage agent reputations
 
-Should you choose to develop inside Arc.js itself, you can find helpful information in the [documentation for Arc.js developers](https://github.com/daostack/arc.js/blob/devDocs/docs/DeveloperDocs.md).
-
-You can find NodeJs scripts that use Arc.js to perform a variety of functions in the [Arc.js-scripts Github repository](https://github.com/daostack/arc.js-scripts).
+## [Subgraph](https://github.com/daostack/subgraph)
+**Subgraph indexes the blockchain data and stores it in postgres database for easy and quich access.** The data store can be queried by GraphQL endpoints. DAOstack subgraph is based on the graphprotocol checkout [TheGraph](https://thegraph.com/docs/quick-start) for more details.
```
[the remark this answers is not in the document]
Comment (shivgupt) [reply README.md]: >     * do we need the whole subgraph code in the starter example ?
> I agree we should not have whole subgraph code in the example. I just saw subgraph is available as npm package. Not really sure how to deploy subgraph using it though. Also, we need to update subgraph deployment scripts to use migration file path set via environment file or argument passed. @ben-kaufman 
>     * where does the migration repo is mention ? how to deploy a single Dao ?
- I have briefly mentioned migration in updated README. Starter template is also an example of deploying a single DAO, should we have more detailed section in top level README for migration and deployment?

**Comment by shivgupt** [comment]
>     * do we need the whole subgraph code in the starter example ?
> I agree we should not have whole subgraph code in the example. I just saw subgraph is available as npm package, not sure how to deploy subgraph using it though. Also, we need to update subgraph deployment scripts to set migration file path via environment var or argument passed. @ben-kaufman 

>     * where does the migration repo is mention ? how to deploy a single Dao ?
> I have briefly mentioned migration in updated README. Starter template is also an example of deploying a single DAO, should we have more detailed section in top level README for migration and deployment?

>     * Another thing to add is that because the subgraph repository is regularly updated the folder should be maintained with the changs.
> Hopefully we will have a better way to deploy subgraph and not need to update the folder, but just have the right version of subgraph library in the `package.json`

File: starter-template/README.md (hunk 2 above)
Comment (ben-kaufman) [reply starter-template/README.md]: I agree, so we can just leave it this way.

**Comment by ben-kaufman** [comment]
The changes look good to me. Regarding using the Subgraph package I will try to make it possible to specify a migration file, but we also need to make sure to update the NPM package as I think right now the version there is very old.

**Pull request merged**